docs/DeveloperGuide.md

```diff
@@ -51,30 +51,63 @@ Step 5. The string at the 0th index is then used in a switch statement, where ea
         
 ### Loading Data
 
-Data is loaded automatically from JSON files in the data folder. This is mainly facilitated through the `TopicsStorage` class which handles accessing the file as well as converting from JSON into `Topic`, `Question` and `Option` objects.
+Data is loaded automatically from JSON files in the data folder. This is mainly facilitated through the `TopicsStorage` 
+class which handles accessing the file as well as converting from JSON into `Topic`, `Question` and `Option` objects.
 
 ![TopicsStorage Class Diagram](./images/TopicsStorage.png)
 
 Given below is an example usage scenario of loading in two topics with two questions each.
 
-When the user launches the app, the main program will initialize a `TopicsStorage` object and call the `load` method which will return a `TopicList` object. The following sequence diagram shows how the load operation works:
+When the user launches the app, the main program will initialize a `TopicsStorage` object and call the `load` method 
+which will return a `TopicList` object. The following sequence diagram shows how the load operation works:
 
 ![TopicsStorage load](./images/TopicsStorage_load.png)
 
+### Class Component 
+
+#### Option and OptionList Class 
+
+The `Option` and `OptionList` classes implements the `Displayable` and `DisplayableList` interfaces respectively. 
+The `Option` object stores one option of a question while the `OptionList` object stores all 4 options of the same 
+question. The class diagram below illustrates the structure of both classes. 
+
+![Option_and_OptionList_Class](./images/Option.png)
+
+### User Interface 
+
+#### Implementation 
+
+The `Ui` class handles all the interactions with the users. It reads the input from the users and prints out replies to 
+the users. An example is provided below to illustrate how the `Ui` class prints out the starting page of the quiz for 
+the user to comprehend. 
+
+![Ui_Printing_Start_Quiz_Page](./images/PrintQuizStartPage.png)
```

Comment:
Not very sure if you need the arrows that do not correspond to a method call or not. Also, the return values should be dashed lines.

Reply:
Okie, I will look into this. Thank you!